### ISSUE TITLE
Fix query behaviour triggered by certain SecuredChild entities

### ISSUE THREAD

**Issue by ppavlidis** [PavlidisLab/Gemma#1024]
There are some operations like ExpressionExperimentService.addFactorValues (used to assign values to continuous factors) which create FactorValues, and generate thousands of ACL SELECT queries in the process

FactorValues are SecuredChildren of ExpressionExperiments, so `this.factorValueService.create( fv )` really shouldn't yield any ACL checks, but it easily generates ~5000 queries. Since this is done for each factorValue being created, it's very slow.

The fundamental issue is not just performance but what must be some underlying bug in what triggers these ACL checks, and it may affect other entities and Service methods. Each FactorValue has an ACL entry, that's fine -- it's just within a Transaction where the secure entity (ExpressionExperiment) is already being checked for permissions, not much else should be happening with respect to ACLs.

**Comment by ppavlidis** [comment]
Still applies in 2.0-alpha; needs focused follow-up. Audited during 2.0-alpha bug triage. The ACL JOIN→EXISTS refactor closed many ACL-query plan issues, but the N+1 ACL SELECT pattern on SecuredChild creation (e.g. `ExpressionExperimentService.addFactorValues`) was not specifically targeted by that work. Worth re-measuring against the current code; the underlying cause is likely an ACL-create path running inside a parent-secured transaction.

**Comment by ppavlidis** [comment]
Investigated as part of the ACL cluster recce; needs a focused perf bench before a targeted fix. Deferred to follow-up.

The N+1 SELECT pattern on SecuredChild creation has changed shape after Phase 3 but may not have improved on the original report:

- The old AOP-based `BaseAclAdvice` ran the ACL setup AFTER the DAO method returned, walking the entity graph by reflection. Each newly-cascaded SecuredChild triggered a `getAclService().readAclById(...)` to locate its parent ACL, multiplied across thousands of FactorValues per addFactorValues call.
- The new `AclEventListener` fires per-entity at flush time and uses a thread-local parent stash plus `resolveParentAcl` to discover parents without re-issuing a SELECT for the parent ACL on each child — IF the stash is populated by the parent's PostInsertEvent earlier in the same flush.

That said, for a flow like `ExpressionExperimentService.addFactorValues` where the EE is ALREADY persistent and only the FactorValues are being inserted, the parent (EE) never fires a PostInsertEvent in this transaction, so its OID is not in the stash. `resolveParentAcl` will fall back to `getSecurityOwner()` then to `discoverParentViaBackRef` — both of which can issue an ACL SELECT per child. `AclEventListener.discoverParentViaBackRef` reads through the ACL service per @ManyToOne back-ref, so the N+1 may persist.

Concrete follow-up when picked up:

1. Set up a perf harness that runs addFactorValues with N=1000 FVs and counts the ACL SELECTs via a Hibernate statistics probe.
2. Confirm/refute that `AclEventListener.discoverParentViaBackRef` is the dominant source.
3. If yes: cache the resolved parent OID for the duration of the flush (the same stash key shape, but populated lazily on the first back-ref discovery).
4. Re-bench.

Out of scope for the cluster commit because the fix needs the measurement first — I do not want to add caching that the bench reveals is unnecessary.

**Comment by ppavlidis** [comment]
Gemma 2 disposition: Won't fix
